packages/polaris-middlewares/src/logger-plugin/polaris-request-listener.ts

```diff
@@ -31,11 +36,30 @@ export class PolarisRequestListener implements GraphQLRequestListener<PolarisGra
             errors: response.errors,
             extensions: response.extensions,
         };
+        const elapsedTime = context.requestStartedTime && Date.now() - context.requestStartedTime;
+        const headers = response.http?.headers;
         this.logger.info(RESPONSE_SENT, context, {
+            eventKind: EventCode.RESPONSE,
             response: loggedResponse,
+            elapsedTime,
+            customProperties: {
+                responseHeaders: headers && [...headers],
+                esc_doc_id: context.logDocumentId,
+                affectedEntitiesCount: this.calculateAffectedEntitiesCount(response),
+            },
         });
     }
 
+    private calculateAffectedEntitiesCount(response: GraphQLResponse) {
+        let affectedEntitiesCount = 0;
+        if (response.data) {
+            Object.values(response.data).map(
```

Comment:
why map? use foreach. map used to map values of array -> more understandable for programmers later on.